gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/RootProvider.java

```diff
@@ -505,6 +513,7 @@ private void csrfHandler(Router router) {
         router.route(PATH_LOGIN_SSO_POST).handler(csrfHandler);
         router.route(PATH_MFA_CHALLENGE).handler(csrfHandler);
         router.route(PATH_MFA_CHALLENGE_ALTERNATIVES).handler(csrfHandler);
+        router.route(PATH_MFA_RECOVERY_CODE).handler(csrfHandler);
```

Comment:
Do we need to set the cspHandler for RecoveryCode route ? (line 531)

Reply:
Recovery code added to cspHandler